File: PERSONALES/Python/Unidad2_Regresion/RLS/RegresionLinearSimple.py
# -*- coding: utf-8 -*-
"""
Created on Thu Jul 15 20:41:30 2021

@author: Icuerec
"""

# Regresión Lineal Simple

# Cómo importar las librerías
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Importar el data set
dataset = pd.read_csv('Unidad2_Regresion/RLS/Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values


# Dividir el data set en conjunto de entrenamiento y conjunto de testing
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Crear modelo de Regresión Lienal Simple con el conjunto de entrenamiento
regression = LinearRegression()
regression.fit(X_train, y_train)

# Predecir el conjunto de test
y_pred = regression.predict(X_test)

# Visualizar los resultados de entrenamiento
plt.scatter(X_train, y_train, color = "red")
plt.plot(X_train, regression.predict(X_train), color = "blue")
plt.title("Sueldo vs Años de Experiencia (Conjunto de Entrenamiento)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo (en $)")
plt.show()

# Visualizar los resultados de test
plt.scatter(X_test, y_test, color = "red")
plt.plot(X_train, regression.predict(X_train), color = "blue")
plt.title("Sueldo vs Años de Experiencia (Conjunto de Testing)")
plt.xlabel("Años de Experiencia")
plt.ylabel("Sueldo (en $)")
plt.show()

File: PERSONALES/R/Unidad2_Regresion/RLS/RegresionLinearSimple.R
#Regresión lineal simple

#Cargamos el archivo
dfSalary = read.csv('Unidad2_Regresion/RLS/Salary_Data.csv')

#Creamos el conjunto de test y el conjunto de entrenamiento
library(caTools)
set.seed(123)

split = sample.split(dfSalary$Salary,SplitRatio = 2/3)
training_set = subset(dfSalary,split==TRUE)
testing_set = subset(dfSalary,split == FALSE)

#Ajustar el modelo de regresión lineal simple con el training_set

regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#Predecir resultados con el set de test (Se tienen que llamar igual las columnas!!!)

y_pred = predict(regressor, newdata = testing_set)

#Visualizar datos y predicción

#install.packages('ggplot2')

library(ggplot2)

ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'orange') +
  geom_line(aes(x = training_set$YearsExperience,
                y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Sueldo vs Años exp. (Conjunto de Entrenamiento)') +
  xlab('Años exp.') + 
  ylab('Sueldo')

